src/chainlit/sync.py

```diff
@@ -21,9 +21,15 @@
 T = TypeVar("T")
 
 
-def run_sync(co: Coroutine[Any, Any, T_Retval]) -> T_Retval:
+def run_sync(co: Coroutine[Any, Any, T_Retval], main_thread=True) -> T_Retval:
+    """Run the coroutine synchronously. If main_thread is True, the coroutine
+    is executed in the main thread, otherwise it is executed in the current
+    thread."""
+
     if threading.current_thread() == threading.main_thread():
         return sync(co)
     else:
+        if not main_thread:
+            return asyncio.run(co)
```

Comment:
why do we use asyncio.run here instead of sync()? Also the main_thread parameter naming might be confusing since we check `threading.current_thread() == threading.main_thread()` in the condition above